source/COMRegistrationFixes/__init__.py

```diff
@@ -11,71 +11,114 @@
+		log.error(f"Error registering {fileName} in a 32-bit context: {e}")
+	else:
+		log.debug(f"Registered {fileName} in a 32-bit context.")
+
+
+def register64bitServer(fileName: str) -> None:
+	"""Registers the COM proxy dll with the given file name, using the 64-bit version of regsvr64.
+	Note: this function is valid while NVDA remains a 32-bit app. Re-evaluate if we move to 64-bit.
 	@param fileName: the path to the dll
 	@type fileName: str
-	@param wow64: If true then the 32 bit (wow64) version of regsvr32 will be used.
-	@type wow64: bool
 	"""
-	regsvr32=os.path.join(sysWow64 if wow64 else system32,'regsvr32.exe')
+	# SysWOW64 provides a virtual directory to allow 32-bit programs to reach 64-bit executables.
+	regsvr32 = os.path.join(SYSNATIVE, "regsvr32.exe")
 	try:
-		subprocess.check_call([regsvr32,'/s',fileName])
+		subprocess.check_call([regsvr32, "/s", fileName])
 	except subprocess.CalledProcessError as e:
-		log.error("Error registering %s, %s"%(fileName,e))
+		log.error(f"Error registering {fileName} in a 64-bit context: {e}")
 	else:
-		log.debug("Registered %s"%fileName)
+		log.debug(f"Registered {fileName} in a 64-bit context.")
+
 
-def applyRegistryPatch(fileName,wow64=False):
+def apply32bitRegistryPatch(fileName: str) -> None:
+	"""Applies the registry patch with the given file name, using 32-bit regedit.
+	Note: this function is valid while NVDA remains a 32-bit app. Re-evaluate if we move to 64-bit.
+	@param fileName: the path to the .reg file
+	@type fileName: str
 	"""
-	Applies the registry patch with the given file name
-	using regedit.
+	if not os.path.isfile(fileName):
+		raise FileNotFoundError(f"Cannot apply 32-bit registry patch: {fileName} not found.")
+	# On 32-bit systems, regedt32 is in System32. On 64-bit systems, SysWOW64 will redirect to 32-bit version.
+	regedit = os.path.join(SYSTEM_ROOT, "System32", "regedt32.exe")
+	try:
+		subprocess.check_call([regedit, "/s", fileName])
+	except subprocess.CalledProcessError as e:
+		log.error(f"Error applying 32-bit registry patch from {fileName} with {regedit}: {e}")
+	else:
+		log.debug(f"Applied 32-bit registry patch from {fileName}")
+
+
+def apply64bitRegistryPatch(fileName: str) -> None:
+	"""Applies the registry patch with the given file name, using 64-bit regedit.
+	Note: this function is valid while NVDA remains a 32-bit app. Re-evaluate if we move to 64-bit.
 	@param fileName: the path to the .reg file
 	@type fileName: str
 	"""
 	if not os.path.isfile(fileName):
-		raise FileNotFoundError(f"Cannot apply registry patch, {fileName} not found.")
-	regedit=os.path.join(sysWow64 if wow64 else systemRoot,'regedit.exe')
+		raise FileNotFoundError(f"Cannot apply 64-bit registry patch: {fileName} not found.")
+	# On 64-bit systems, SysWOW64 provides 32-bit apps with a virtual directory to reach 64-bit executables.
+	# For hard coded historical compatibility, regedt32.exe is still the name of the 64-bit app.
+	regedit = os.path.join(SYSNATIVE, "regedt32.exe")

```

Comment:
Have you considered using `reg.exe` rather than `regedit` here? As discussed in #11494 the latter always returns 0 as its exit code even on failure and since you're refactoring this code anyway it makes sense to solve more known problems IMO.

Reply:
@lukaszgo1 I didn't use reg.exe for the simple fact that I was unaware of it. @jcsteh recommended regedt32.exe, so that's what I went with. I have updated now to use reg.exe, thanks.
So far I left it loading the values from the .reg file still, instead of directly including the keys in source.

Regarding a 32-bit virtual test machine: I would need to switch to a Win 10 Pro installation on my work machine in order to setup a proper virtual environment. There are various reasons that would be highly inconvenient right now, although it is on my list.